sycl/test/array_param/array-kernel-param-run.cpp

```diff
@@ -0,0 +1,250 @@
+// This test checks kernel execution with array kernel parameters.
+
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.out
+// RUN: env SYCL_DEVICE_TYPE=HOST %t.out
+// RUN: %CPU_RUN_PLACEHOLDER %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+// RUN: %ACC_RUN_PLACEHOLDER %t.out
+
+#include <CL/sycl.hpp>
+#include <iostream>
+
+using namespace cl::sycl;
+
+constexpr size_t c_num_items = 100;
+range<1> num_items{c_num_items}; // range<1>(num_items)
+
+// Change if tests are added/removed
+static int testCount = 5;
+static int passCount;
+
+template <typename T>
+static bool verify_1D(const char *name, int X, T A, T A_ref) {
+  int ErrCnt = 0;
+
+  for (int i = 0; i < X; i++) {
+    if (A_ref[i] != A[i]) {
+      if (++ErrCnt < 10) {
+        std::cout << name << " mismatch at " << i << ". Expected " << A_ref[i]
+                  << " result is " << A[i] << "\n";
+      }
+    }
+  }
+
+  if (ErrCnt == 0) {
+    return true;
+  }
+  std::cout << "  Failed. Failure rate: " << ErrCnt << "/" << X << "("
+            << ErrCnt / (float)X * 100.f << "%)\n";
+  return false;
+}
+
+template <typename T>
+void init(T &A, int value, int increment) {
+  for (int i = 0; i < c_num_items; i++) {
+    A[i] = value;
+    value += increment;
+  }
+}
+
+bool test_one_array(queue &myQueue) {
+  int input1[c_num_items];
+  int output[c_num_items];
+  int ref[c_num_items];
+  init(input1, 1, 1);
+  init(output, 51, 1);
```

Comment:
is this needed?

Reply:
Yes, the verify_1D checks that the original values in output (51...) are over-written with new values (2...).